internal/app/machined/pkg/system/runner/cri/cri_test.go

```diff
@@ -0,0 +1,338 @@
+
+	suite.Require().NoError(r.Open(context.Background()))
+	defer func() { suite.Assert().NoError(r.Close()) }()
+
+	suite.Assert().NoError(r.Run(MockEventSink))
+	// calling stop when Run has finished is no-op
+	suite.Assert().NoError(r.Stop())
+}
+
+func (suite *CRISuite) TestRunTwice() {
+	r := crirunner.NewRunner(&userdata.UserData{}, &runner.Args{
+		ID:          "runtwice",
+		ProcessArgs: []string{"/bin/sh", "-c", "exit 0"},
+	},
+		runner.WithLogPath(suite.tmpDir),
+		runner.WithContainerImage(suite.imageRef),
+		runner.WithContainerdAddress(suite.containerdAddress),
+	)
+
+	suite.Require().NoError(r.Open(context.Background()))
+	defer func() { suite.Assert().NoError(r.Close()) }()
+
+	// running same container twice should be fine
+	// (checks that containerd state is cleaned up properly)
+	for i := 0; i < 2; i++ {
+		suite.Assert().NoError(r.Run(MockEventSink))
+		// calling stop when Run has finished is no-op
+		suite.Assert().NoError(r.Stop())
+
+		// TODO: workaround containerd (?) bug: https://github.com/docker/for-linux/issues/643
+		time.Sleep(100 * time.Millisecond)
+	}
+}
+
+func (suite *CRISuite) TestPodCleanup() {
+	// create two runners with the same  ID
+	//
+	// open first runner, but don't close it; second runner should be
+	// able to start the pod by cleaning up pod created by the first
+	// runner
+	r1 := crirunner.NewRunner(&userdata.UserData{}, &runner.Args{
+		ID:          "cleanup1",
+		ProcessArgs: []string{"/bin/sh", "-c", "exit 1"},
+	},
+		runner.WithLogPath(suite.tmpDir),
+		runner.WithContainerImage(busyboxImage),
+		runner.WithContainerdAddress(suite.containerdAddress),
+	)
+
+	suite.Require().NoError(r1.Open(context.Background()))
+
+	r2 := crirunner.NewRunner(&userdata.UserData{}, &runner.Args{
+		ID:          "cleanup1",
+		ProcessArgs: []string{"/bin/sh", "-c", "exit 0"},
+	},
+		runner.WithLogPath(suite.tmpDir),
+		runner.WithContainerImage(busyboxImage),
+		runner.WithContainerdAddress(suite.containerdAddress),
+	)
+	suite.Require().NoError(r2.Open(context.Background()))
+	defer func() { suite.Assert().NoError(r2.Close()) }()
+
+	suite.Assert().NoError(r2.Run(MockEventSink))
+	// calling stop when Run has finished is no-op
+	suite.Assert().NoError(r2.Stop())
+}
+
+func (suite *CRISuite) TestRunLogs() {
+	r := crirunner.NewRunner(&userdata.UserData{}, &runner.Args{
+		ID:          "logtest",
+		ProcessArgs: []string{"/bin/sh", "-c", "echo -n \"Test 1\nTest 2\n\""},
+	},
+		runner.WithLogPath(suite.tmpDir),
+		runner.WithContainerImage(busyboxImage),
+		runner.WithContainerdAddress(suite.containerdAddress),
+	)
+
+	suite.Require().NoError(r.Open(context.Background()))
+	defer func() { suite.Assert().NoError(r.Close()) }()
+
+	suite.Assert().NoError(r.Run(MockEventSink))
+
+	logContents := suite.getLogContents("logtest.log")
+
+	suite.Assert().Contains(string(logContents), "Test 1\n")
+	suite.Assert().Contains(string(logContents), "Test 2\n")
+}
+
+// func (suite *CRISuite) TestStopFailingAndRestarting() {
```

Comment:
Is this commented on purpose?

Reply:
yes, I copy-pasted it from containerd tests, but I can't enable it yet, as CRI runner doesn't support mounts, so my plan is to enable it once mounts are supported